### ISSUE TITLE
Update react dependency

### ISSUE THREAD

**Issue by crowbait** [jepiqueau/react-sqlite-hook#1]
I tried to install this hook using @latest, npm produces:

```
npm install react-sqlite-hook@latest          
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: RandomChat@0.0.1
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"~17.0.2" from react-sqlite-hook@3.0.1
npm ERR! node_modules/react-sqlite-hook
npm ERR!   react-sqlite-hook@"3.0.1" from the root project
```

Although I'm pretty sure it will work just fine using `--force`, it would be nice to update the dependencies.

**Comment by crowbait** [comment]
Update: even with `--force`, it does indeed NOT work.
Typescript complains about not being able to find the package although it is clearly there. I guess the dependency issue can't simply be solved with force.
`--legacy-peer-deps` seems to work for the moment; will keep you updated.

**Comment by crowbait** [comment]
Update: `--legacy-peer-deps` does indeed _not_ work. While the package installs and Typescript doesn't complain anymore, `ionic build` fails.

**Comment by jepiqueau** [comment]
@Traxx2012 is your app is running with react@18.2.0 ? I did not test react@18.2.0 this is why i have put a peer-dependency of 17.0.2.
Is your app a pure  React app or a Ionic/React app ?

**Comment by crowbait** [comment]
@jepiqueau Hi! First of all, amazing work you're doing here. Spent the last hour realizing just how much work you pumped into this whole extension and I'm terribly impressed.

The app is an Ionic/React app, running the following dependencies:
```
"@ionic/react": "^6.0.0",
"react": "^18.2.0"
```

**Comment by jepiqueau** [comment]
@Traxx2012 thanks for your comments. yes i gave a lot of work and time to make it. And developers are help me with tests and issues to make it what it is today. I did not see that when creating a Ionic/React app   with ionic start ... the Ionic team has updated to react@18.2.0 last time i looked it was still 17.0.2 that is the reason i keep the peer-dependency to react17.0.2

**Comment by crowbait** [comment]
@jepiqueau No worries, I understand how time limitations affect projects like this.
I wanted to let you know that I've tried your release from ~1h ago. I don't know if you even intended to see to this issue, but just in case I wanted to let you know that it didn't solve it.
So `ionic build` still doesn't compile and fails.

**Comment by jepiqueau** [comment]
@Traxx2012 the intent of the release one hour ago what to fix the issue#324 not that one

**Comment by jepiqueau** [comment]
@Traxx2012 this is done in release 3.1.0 but when i test it in react-sqlite-app-starter 
if i keep 

```
    ReactDOM.render(
      <React.StrictMode>
        <App /> 
      </React.StrictMode>,
      document.getElementById('root')
    );
```
it works
but when i replace ReactDOM.render with 

```
const container = document.getElementById('root');
const root = createRoot(container!);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
```
it does not work seems there is an issue with the listener.remove() which i could not find yet the reason for. If you have an idea you are welcome

**Comment by crowbait** [comment]
My friend, I'm sorry but this is my first project with React... I know what a listener is, but man, if you say there is some problem I'll just have to believe you and wish you the best of luck in finding it...
I really wished I could contribute a bit to your awesome - and terrifyingly complex - project. I'll have a look at the code, but please don't expect me to be terribly successful...

**Comment by crowbait** [comment]
The issue seems fixed with 3.1.1 . Package installs fine and `ionic build` doesn't complain anymore.
I didn't have time to deeply test this yet, but I'm closing this issue because I'm sure that if anything else arises, it's likely to be a different issue.

Again, thank you for all that work you're doing!

**Comment by jepiqueau** [comment]
@Traxx2012 yes it is fixed in 3.1.1. Good luck for your development
